### DISCUSSION TITLE
Intellij setup and flex sdk overlay

### DISCUSSION THREAD

**Discussion by dviel** [airsdk/Adobe-Runtime-Support#1844] [General]
Dear all,

I'm moving from Flash builder to Intellij 🤯, but I'm struggling to setup my dev env.
Do you have any online tutorial to do that correctly ?

Many thanks

**Comment by teotigraphix** [comment]
I use Intellij everyday, just ask specific questions you are stuck on and I can try to help.

**Reply by toysonlocation** [reply]
I have a specific question, maybe. I'm working on an inherited FLEX codebase and am getting this error:

```
/Volumes/Tub/Work/Projects/gen/app/src/main/com/test/app/story/tutorial/view/Hint.mxml
[app-dev (module app)]: Implementation of interface 'flash.events:IEventDispatcher' cannot be added to class 'Hint', due to collision with existing method or property 'addEventListener'.
```

It actually reports that error 4 times. Also, it does it for every mxml file in the project. I've created a Flex/AIR combo (Apache Flex 4.16.1 with AIR-for-Flex 821 overlaid, and also overlaid the AIR-for-pure-as3 821/frameworks/asc-support.swc file which was missing from AIR-for-Flex), but the above error(s) keeps appearing when I try to build. ANY advice would be greatly appreciated.

**Reply by ajwfrost** [reply]
This sounds like an issue with that "Hint" class - is there any definition of a function called `addEventListener` in it? You'd need to have it specified as an override, since most (all?) MXML-based view classes will derive from IEventDispatcher at some level... 

Although .. if its' happening for every MXML file, then it perhaps implies a difference in the libraries (e.g. airglobal.swc) which is an interesting one; we really shouldn't have touched IEventDispatcher though!

I'm interested by your mention of asc-support.swc; this shouldn't be required for a Flex project with the original compiler. Can you open a terminal window, go to the AIR SDK folder into "bin", and run `./mxmlc -version`?

thanks

**Comment by toysonlocation** [comment]
Hi @ajwfrost - thanks for the reply. Here is the output from mxmlc:

Version 4.16.1 build 20171115

There actually isn't a single addEventlistener function defined in the entire project (override or otherwise). Could it be a Starling conflict?

**Reply by ajwfrost** [reply]
Not sure it's going to be related to Starling, assuming your Hint class is just a normal (Flex) component. Have you tried a basic/empty Flex application to see if that works?

What's the type of object here? i.e. 'Hint' derives from what? Any chance of a simple/stripped-back demonstrator of the issue?

Actually thinking about it .. if you have an object that extends any Starling display object, then it wouldn't be possible to add the flash.events.IEventDispatcher interface -> but also, it shouldn't be happening i.e. nothing should be trying to implement that interface if you're using Starling components...

**Reply by toysonlocation** [reply]
OK, I've created a totally new project, and am still getting the error about IEventDispatcher collisions. I also noticed that I am having to fully qualify starling.events.Event in the button callback signature, as in the following pic, whereas the demo code is simply `(event:Event)`. This is all the code in the entire project, in Main.mxml:

![image](https://user-images.githubusercontent.com/39204807/166114324-bfc889fe-dbb5-4520-92b4-16d82584e1b3.png)

But I am getting the same set of 4 errors on the mxml file as I am with the large project (also on all mxml files):

![image](https://user-images.githubusercontent.com/39204807/166114437-2ff7be53-d199-479e-97c0-1e1b229026f7.png)

I've included the Feathers 4.1.1/frameworks/libs, which includes starling.swc:

![image](https://user-images.githubusercontent.com/39204807/166114191-ae3b7ab1-fd58-4e11-81af-244724fdf884.png)

Apache Flex 4.16.1
Harman AIR 33.1.1.821 (for Flex)
Feathers 4.1.1 (with bundled starling.swc in FEATHERS_SDK/frameworks/libs/)

![image](https://user-images.githubusercontent.com/39204807/166114624-5284054e-b084-4e68-a1bb-4daf2bf1b509.png)
![image](https://user-images.githubusercontent.com/39204807/166114651-67cd77ee-3f96-42a5-b12a-000cd0d128cd.png)

**Reply by toysonlocation** [reply]
Here is the IntelliJ project file
[lib_test.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/8597453/lib_test.zip)

**Reply by ajwfrost** [reply]
How are you creating the SDK there? I think there could be a problem if you've got a "normal" Flex SDK and are then trying to use the Feathers components just by linking to the SWC file. I think the compiler is assuming that any component will need to inherit from IEventDispatcher but that's not the case for the Feathers Application object which derives from a Starling Sprite object. Plus of course there would need to be some other bootstrap code here, you can't just have an application that directly derives from a Starling Sprite, it would need the Starling core class to be created and started...

So I would suggest to start with the Feathers SDK downloaded via their manager utility, and then add the Harman AIR SDK for Flex over the top of this?

thanks

**Reply by toysonlocation** [reply]
Yes, it turned out to me my misunderstanding of how FLEX/Feathers/AIR SDKs (and hierarchies) work. I figured out that indeed it was what you suggested in the last sentence: to overlay the Harman AIR for FLEX SDK onto the Feathers SDK. That fixed all things. I was using Apache FLEX overlaid with Harman and including Feathers.

Thanks for your help, and sorry for the wheel spinning.

**Reply by ajwfrost** [reply]
Great - yes, I think what Josh has done with that Feathers SDK is very clever and goes a bit beyond the normal "add a few components", it seems there's  a very customised build of the ActionScript/MXML compiler in there too!

thanks

**Comment by toysonlocation** [comment]
Hi @ajwfrost UPDATE: 
I thought the 4 errors were all the same, but actually it looks most of the flash.events.IEventDispatcher interface is being thrown. Only `willTrigger()` is not being hit. None of the functions, and also not willTrigger, are redefined or being overridden in the project. As before, all mxml files are throwing these:

```
[app-dev (module app)]: Implementation of interface 'flash.events:IEventDispatcher' cannot be added to class 'HintContent', due to collision with existing method or property 'addEventListener'
[app-dev (module app)]: Implementation of interface 'flash.events:IEventDispatcher' cannot be added to class 'HintContent', due to collision with existing method or property 'removeEventListener'
[app-dev (module app)]: Implementation of interface 'flash.events:IEventDispatcher' cannot be added to class 'HintContent', due to collision with existing method or property 'dispatchEvent'
[app-dev (module app)]: Implementation of interface 'flash.events:IEventDispatcher' cannot be added to class 'HintContent', due to collision with existing method or property 'hasEventListener'
```
